security/tls-authorization/README.adoc

```diff
@@ -0,0 +1,226 @@
+**Summary:** Use SSL/TLS certificates to verify Hot Rod client identities and encrypt network traffic to the server. This tutorial also demonstrates how to perform client certificate authentication and use security authorization with client certificates.
+
+== Building instructions and preparation
+
+. `$ mvn clean package`
+. Build the example certificates and keystores with the following scripts or directly with the Java keytool commands:
+* `create_unsigned_certificates.sh`
+* `create_signed_certificates.sh`
++
+These scripts create keystores and trust stores in the project directory.
+
+. Make sure you are using Java 11 at a minimum.
++
+Infinispan Server requires at least Java 11.
+Additionally the example keystores and trust stores in this tutorial use `PKCS12`, which is the recommended standard and the default from Java 11.
+. Download and start at least one Infinispan Server instance.
+
+== Unsigned Server certificates
+
+. Copy `server-keystore.pfx` to `$ISPN_HOME/server/conf`.
+. Add the `server-identities` configuration to the Infinispan Server security realm.
++
+Note that the default properties realm is removed.
++
+[source,xml]
+----
+<security-realm name="default">
+  <server-identities>
+    <ssl>
+      <keystore path="server-keystore.pfx"
+                keystore-password="secret"
+                alias="infinispan-server"/>
+    </ssl>
+  </server-identities>
+</security-realm>
+----
++
+. Create a cache named `secured` with any configuration.
++
+[source,xml]
+----
+<distributed-cache name="secured" mode="SYNC">
+  <encoding media-type="application/x-protostream"/>
+</distributed-cache>
+----
++
+. Run the client.
++
+[source]
+----
+$ export TEST=SIMPLE && mvn exec:exec
+----
+
+=== Unsigned Server certificates with client authorization
+
+. Copy `client-keystore.pfx` to `$ISPN_HOME/server/conf`.
+. Configure Infinispan Server to authenticate client certificates by adding `require-ssl-client-auth="true"` to the `endpoints` configuration.
++
+[source,xml]
+----
+<endpoints socket-binding="default"
+           security-realm="default"
+           require-ssl-client-auth="true"/>
+----
++
+. Add the `server-identities` configuration to the Infinispan Server security realm.
+. Include the `<truststore-realm/>` to authenticate certificates, which requires all public client certificates in the trust store.
++
+Note that the default properties realm is removed.
++
+[source,xml]
+----
+<security-realm name="default">
+  <server-identities>
+    <ssl>
+      <keystore path="server-keystore.pfx"
+                keystore-password="secret"
+                alias="infinispan-server"/>
+      <truststore path="client-truststore.pfx"
+                  password="ClientSecret"/>
+    </ssl>
+  </server-identities>
+  <truststore-realm/>
+</security-realm>
+----
++
+. Update the security authorization configuration for Infinispan Server to use the Common Name role mapper and define two client roles and associated permissions.
```

Comment:
@wfink What do you think about removing the section about authz here? 

You could put the security authz steps at the end of the Signed Certificate part of the tutorial only to avoid some duplication.

If you run the clients after updating the server configuration with the client trust store then it shows client certificate authentication more clearly. I think it would reduce complexity for this tutorial to keep authz to last.